particle/development/using-demos.md

```diff
@@ -0,0 +1,140 @@
+so you should never do work on them that you expect to see on a production app,
+such as a Drupal theme.
+
+Creating a demo folder should be one of the first steps in the development of any
+new component, and updating it as development goes on should be a continuous task.
+
+## Components and Creation
+
+Demo folders are one of the options for creation when using Yeoman to [generate a new
+component](../architecture/component-structure.md#generating-a-component). The
+Yeoman generator creates four files by default in this folder, which describe the
+general Phase2 way to demonstrate how to use a new component:
+
+### index.js
+
+Describes to Webpack the files to include when generating the Pattern Lab demos.
+This most likely will only need to be touched when you are creating multiple demo
+pages for a single component.
+
+### components.twig
+
+{% hint style="info" %}
+Note the pluralization of the component name, which indicates multiple examples
+of the component!
+{% endhint}
+
+This should serve as an example twig file that shows an inclusion of the new pattern,
+placed with example variables and layouts so that it will appear like an intended
+component when viewed in the demos.
+
+The default .twig file places a standard Twig-style `include` of your new component,
+but you can (and should) modify this page to meet your design system better. Look
+at the default patterns for good examples of how to create a demo template.
+
+### component.md
+
+An additional readme that may describe how to use your demo, its implications, or
+others. By default the variables that a component requires are often documented
+here, in addition to in the component's `_component.twig` file.
+
+### component.yml
+
+An optional way to override the data in `/_data/data.yml`, to display something
+specific in your demo. These data structures are only included in their respective
+page's demos when rendering the demo's template, so they are safely separated from
+other pattern demos.
+
+{% hint style="info" %}
+If you would like to create demo data to be global between all patterns, add
+the variables to a Yaml file in the `/_data` folder, such as `/_data/data.yml`.
+{% endhint %}
+
+For more information about how Pattern Lab expects variables to be added for use
+in patterns, see the [Pattern Lab docs](https://patternlab.io/docs/data-overview.html).
+
+## Placeholders
+
+Particle includes several ways you can generate defaults for your testing, thus
+supplanting the need to include placeholder images, or generate long strings of
+text.
+
+### Text with Faker
+
+Particle includes the [Faker plugin of Pattern Lab](https://github.com/pattern-lab/plugin-php-faker),
+thus letting one use the [Node.JS Faker API](https://www.npmjs.com/package/faker)
+in order to generate random data. This can be used for anything from names to dates
+to lorem ipsum.
+
+To use Faker, call the Faker object from within your demo's .yml file, with the
```

Comment:
I'm 90% sure Faker only works from within the global _data/ .yml files, not a given component's demo. But you can still loop through global data from a component or component demo

Reply:
@tjheffner, You are correct! Just tested it. Updated the repo in the latest push.